helix-core/src/main/java/org/apache/helix/task/TaskDriver.java

```diff
@@ -77,6 +76,11 @@
   /** Default time out for monitoring workflow or job state */
   private final static int DEFAULT_TIMEOUT = 5 * 60 * 1000; /* 5 mins */
 
+  /** The illegal job states for the jobs to accept new task */
+  private static Set<TaskState> illegalJobStatesForTaskAddition = new HashSet<>(
```

Comment:
nit, private final static... please
Also, in this case, the var name should be ILLEGAL_JOB_STATES_FOR_TASK_ADDITION (or TASKS_MODIFICATION?)

Reply:
Done.